### PR TITLE
Remove duplicate lock DocumentState.s_syntaxTreeToIdMapLock

### PR DESCRIPTION
**Pull request by sharwell**
This lock is only being used to protect access to an instance which contains
internal synchronization.

### CODE REVIEW

File: src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs [hunk 1]
```diff
@@ -693,32 +693,33 @@ public bool TryGetTopLevelChangeTextVersion(out VersionStamp version)
...
-                }
-                else
-                {
-                    s_syntaxTreeToIdMap.Add(tree, id);
-                }
+                Contract.ThrowIfFalse(existingId == id);
+                return;
+            }
+
+            // ConditionalWeakTable is a concurrent map, but doesn't offer a TryAdd method. Use try/catch to behave like
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs]: Any reason not to use [GetValue](https://msdn.microsoft.com/en-us/library/dd413870(v=vs.110).aspx)? Last time I did a pattern like this @KirillOsenkov never let me live it down.

Comment (sharwell) [inline src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs]: No, a definite oversight on my part.

**Comment by sharwell** [comment]
@Pilchie for 15.7 pre-ask mode

**Comment by Pilchie** [comment]
Approved

**Pull request merged**